### PR TITLE
api: Add list_links and get_link api calls

### PR DESCRIPTION
**Pull request by zombah**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: doc/API/Switching.md [hunk 1]
```diff
@@ -69,3 +69,81 @@ Output:
...
+Route: `/api/v0/resources/links`
+
+Input:
+
+  - hostname = Either the devices hostname or id
+
+Example:
+```curl
+curl -H 'X-Auth-Token: YOURAPITOKENHERE' https://librenms.org/api/v0/resources/links
+curl -H 'X-Auth-Token: YOURAPITOKENHERE' https://librenms.org/api/v0/resources/links?hostname=host.example.com
```
Comment (laf) [inline doc/API/Switching.md]: I wonder if this route should be `/api/v0/devices/host.example.com/links`?

Comment (zombah) [reply doc/API/Switching.md]: Yes resonable, i'll prepare it in follow-up commit.

Comment (laf) [reply doc/API/Switching.md]: Can you do it in this one, no point in releasing something that will change, especially if people start using it.

Comment (zombah) [reply doc/API/Switching.md]: Yes sure in this PR. Just added as separate call.

**Pull request merged**